docs/streams/upgrade-guide.html

```diff
@@ -117,6 +117,14 @@ <h3><a id="streams_api_changes_300" href="#streams_api_changes_300">Streams API
     <p>
         We removed the default implementation of <code>RocksDBConfigSetter#close()</code>.
     </p>
+
+    <p>
+        The public <code>topicGroupId</code> and <code>partition</code> fields on TaskId have been deprecated and replaced with getters, please migrate to using the new <code>TaskId.subtopology()</code>
+        and <code>TaskId.partition()</code> APIs instead. Also, the <code>TaskId#readFrom</code> and <code>TaskId#writeTo</code> methods have been deprecated and will be removed, as they were never intended
+        for public use to begin with. Finally, we have deprecated the <code>TaskMetadata.taskId()</code> method as well as the <code>TaskMetadata</code>constructor. These have been replaced with APIs that
+        better represent the task id as an actual <code>TaskId</code> object instead of a String. Please migrate to the new <code>TaskMetadata#getTaskId</code> method and the new constructor which accepts
```

Comment:
`and the new constructor`

I thought that the constructor was never intended for public usage. Should we not hide the new constructor?

Reply:
True. I'll remove this note about it